crates/federated-server/src/server/gateway/graph.rs

```diff
@@ -0,0 +1,97 @@
+use std::path::PathBuf;
+
+use gateway_config::Config;
+use runtime::trusted_documents_client::{Client, TrustedDocumentsEnforcementMode};
+use ulid::Ulid;
+
+use crate::{
+    AccessToken, ObjectStorageResponse,
+    server::trusted_documents_client::{TrustedDocumentsClient, TrustedDocumentsClientConfig},
+};
+
+pub struct Graph {
+    pub federated_sdl: String,
+    pub version_id: Option<Ulid>,
+    pub trusted_documents: Option<Client>,
+    pub current_dir: Option<PathBuf>,
+}
+
+#[derive(Clone)]
+pub enum GraphDefinition {
+    /// Response from object storage.
+    ObjectStorage {
+        response: ObjectStorageResponse,
+        object_storage_base_url: url::Url,
+    },
+    /// Response from static file.
+    Sdl(Option<PathBuf>, String),
+}
+
+impl GraphDefinition {
+    /// Converts a `GraphDefinition` into a `Graph`.
+    ///
+    /// This method processes the graph definition based on its type:
+    ///
+    /// - For `ObjectStorage` variants, it returns a graph with potential trusted documents support
+    /// - For `Sdl` variants, it returns a graph without trusted documents
+    ///
+    /// Trusted documents will only be enabled for `ObjectStorage` variants when both an access
```

Comment:
I don't know if that was the logic before this refactoring, but the logic we would want to land on is making trusted documents work for graphs defined from SDL, as long as the graph ref is present (just like for telemetry)

Reply:
Discussed on slack. This is not a problem introduced by this PR, let's fix separately. Issue: GB-9274